### PR TITLE
[terra-framework] Allow hover styles to apply only on non-touch devices.

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
This PR resolves the issue by making use of hover and pointer interaction media features introduced in Media Queries Level 4 so that the hover styles are applied only on non-touch devices whose primary input is a fine pointer device such as a mouse or a stylus

### Deployment Link

https://terra-framework-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by mjhenkes** [comment]
How are we testing this change? I'm not seeing any differences in the wdio screenshots?

**Comment by emilyrohrbough** [comment]
@saket2403 why did the screenshots change with this media query? Are we being too specific with `and (pointer: fine)` included in the query?

**Comment by saket2403** [comment]
Changes reverted to use media queries and wdio tests removed for hover styles.

File: packages/terra-application-links/src/tabs/ApplicationTabs.module.scss [hunk 1]
```diff
@@ -172,11 +172,13 @@
       word-break: normal;
       word-wrap: normal;
     }
-
-    &:hover,
-    &.is-hovered {
-      .tab-inner {
-        background-color: var(--terra-application-links-tabs-tab-hover-background-color, rgba(0, 0, 0, 0.15));
+    /* stylelint-disable max-nesting-depth */
```
Comment (emilyrohrbough) [inline packages/terra-application-links/src/tabs/ApplicationTabs.module.scss:175]: does this need to be re-enabled or can we only disable the next line?

File: packages/terra-application-navigation/src/extensions/Extension.module.scss [hunk 2]
```diff
@@ -9,10 +9,13 @@
     justify-content: center;
     outline: none;
     position: relative;
-
-    &:hover {
-      .extension-inner {
-        color: var(--terra-application-navigation-extension-hover-color, #fff);
+    
+    /* stylelint-disable max-nesting-depth */
```
Comment (emilyrohrbough) [inline packages/terra-application-navigation/src/extensions/Extension.module.scss:13]: does this need to be re-enabled or can we only disable the next line?

File: packages/terra-application-navigation/src/extensions/ExtensionRollup.module.scss [hunk 3]
```diff
@@ -10,9 +10,12 @@
     outline: none;
     position: relative;
 
-    &:hover {
-      .extension-rollup-inner {
-        color: var(--terra-application-navigation-extension-rollup-hover-color, #fff);
+    /* stylelint-disable max-nesting-depth */
```
Comment (emilyrohrbough) [inline packages/terra-application-navigation/src/extensions/ExtensionRollup.module.scss:13]: same questions as above

File: packages/terra-application-links/src/tabs/ApplicationTabs.module.scss (hunk 1 above)
Comment (saket2403) [reply packages/terra-application-links/src/tabs/ApplicationTabs.module.scss]: Disabled for next line only 0c9bba3

File: packages/terra-application-navigation/src/extensions/Extension.module.scss (hunk 2 above)
Comment (saket2403) [reply packages/terra-application-navigation/src/extensions/Extension.module.scss]: Disabled for next line only 0c9bba3

File: packages/terra-application-navigation/src/extensions/ExtensionRollup.module.scss (hunk 3 above)
Comment (saket2403) [reply packages/terra-application-navigation/src/extensions/ExtensionRollup.module.scss]: Disabled for next line only 0c9bba3

**Comment by neilpfeiffer** [comment]
Tested: PR #1095
Resolves: [Terra-Core: Issue #2927](https://github.com/cerner/terra-core/issues/2927)
Relates to: [Terra-Core: PR #2932](https://github.com/cerner/terra-core/pull/2932) and [Terra-Clinical: PR #652](https://github.com/cerner/terra-clinical/pull/652)

**Reviewed:**
Manually reviewed the hover states for the components affected are correct and functioning as intended:
- terra-application
- (legacy: application-navigation & application-utility)
- brand-footer
- date-input
- date-picker
- menu
- tabs

\
\
Verified by @neilpfeiffer, ready to merge after engineering review.

**Pass** |
-----------|
UX Review|

**Pull request merged**